Microsoft.OperationalInsights: 2025-02-01 New stable version

roizhv22 opened the pull request:
# ARM (Control Plane) API Specification Update Pull Request 

> [!TIP]
> Overwhelmed by all this guidance? See the `Getting help` section at the bottom of this PR description.

## PR review workflow diagram

Please understand this diagram before proceeding. It explains how to get your PR approved & merged.

![spec_pr_review_workflow_diagram](https://github.com/Azure/azure-rest-api-specs/assets/4429827/5bb5e7ce-8aff-4dbb-a3f8-0d9b68fef5b1)

## Purpose of this PR

What's the purpose of this PR? Check the specific option that applies. This is **mandatory**!

  - [ ] New resource provider.
  - [x] New API version for an existing resource provider. (If API spec is not defined in TypeSpec, the PR should have been created in adherence to [OpenAPI specs PR creation guidance](https://aka.ms/azsdkdocs/createopenapispec)).
  - [ ] Update existing version for a new feature. (This is applicable only when you are revising a private preview API version.)
  - [ ] Update existing version to fix OpenAPI spec quality issues in S360.
  - [ ] Convert existing [OpenAPI spec to TypeSpec spec](https://aka.ms/typespec/conversion) (do not combine this with implementing changes for a new API version).
  - [ ] Other, please clarify:
    - _edit this with your clarification_

## Due diligence checklist

To merge this PR, you **must** go through the following checklist and confirm you understood 
and followed the instructions by checking all the boxes:

- [x] I confirm this PR is modifying Azure Resource Manager (ARM) related specifications, and not data plane related specifications.
- [x] I have reviewed following [Resource Provider guidelines](https://aka.ms/rpguidelines), including
  [ARM resource provider contract](https://aka.ms/azurerpc) and
  [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md) (estimated time: 4 hours).  
  I understand this is required before I can proceed to the diagram Step 2, "ARM API changes review", for this PR.

## Additional information

<details>
<summary> Viewing API changes</summary>

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

</details>
<details>
<summary>Suppressing failures</summary>

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[suppressions guide](https://aka.ms/azsdk/pr-suppressions) to get approval.

</details>

## Getting help

- First, please carefully read through this PR description, from top to bottom. Please fill out the `Purpose of this PR` and `Due diligence checklist`.
- If you don't have permissions to remove or add labels to the PR, request `write access` per [aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories](https://aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories)
- To understand what you must do next to merge this PR, see the `Next Steps to Merge` comment. It will appear within few minutes of submitting this PR and will continue to be up-to-date with current PR state.
- For guidance on fixing this PR CI check failures, see the hyperlinks provided in given failure 
  and https://aka.ms/ci-fix.
- For help with ARM review (PR workflow diagram Step 2), see https://aka.ms/azsdk/pr-arm-review.
- If the PR CI checks appear to be stuck in `queued` state, please add a comment with contents `/azp run`.
  This should result in a new comment denoting a `PR validation pipeline` has started and the checks should be updated after few minutes.
- If the help provided by the previous points is not enough, post to https://aka.ms/azsdk/support/specreview-channel and link to this PR.

Commit:
Copy files from stable/2023-09-01

Copied the files in a separate commit.
This allows reviewers to easily diff subsequent changes against the previous spec.

Commit:
Update version to stable/2025-02-01

Updated the API version from stable/2023-09-01 to stable/2025-02-01.

Commit:
Added tag for 2025-02-01 in readme file

roizhv22 committed:
Add replication properties to workspace and cluster contract

Add new properties payload for cluster and workspace

roizhv22 committed:
Add new examples

new examples with replication and failover for workspace

roizhv22 committed:
fix

azure-sdk commented:
**API change check**

APIView has identified API level changes in this PR and created following API reviews.

[common](https://apiview.dev/Assemblies/Review/da811d14b94f40769492b2ede877e48b?revisionId=6fb6bf44553a46108ffeac02f1445d86)

roizhv22 committed:
run npm prettier

roizhv22 committed:
fix required

fix required

roizhv22 committed:
fix example and remove not in use 201

The 201 is not in use. Otherwise fix the exmaple's responses

roizhv22 committed:
set required to on

avoid breaking changee

roizhv22 committed:
add Canceled and fix 202 response

Fix

roizhv22 committed:
Merge branch 'main' into roizehavi/OiNewStable

roizhv22 committed:
Merge branch 'main' into roizehavi/OiNewStable

roizhv22 committed:
Merge branch 'main' into roizehavi/OiNewStable

roizhv22 committed:
Merge branch 'main' into roizehavi/OiNewStable

roizhv22 commented:
@microsoft-github-policy-service agree company="Microsoft"

roizhv22 committed:
add missing x-ms-mutability

roizhv22 committed:
Merge branch 'roizehavi/OiNewStable' of https://github.com/roizhv22/azure-rest-api-specs into roizehavi/OiNewStable

razvanbadea-msft commented on specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2025-02-01/Workspaces.json:
```diff
@@ -0,0 +1,890 @@
+      "type": "object",
+      "description": "Workspace replication properties.",
+      "properties": {
+        "location": {
```
>location [](http://example.com/codeflow?start=9&length=8)

you should rename this for example to azureLocation as this is a top level property used in the property bag

ms-bogdan-rotaru replied:
This is not addressed.

razvanbadea-msft commented on specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2025-02-01/Workspaces.json:
```diff
@@ -0,0 +1,890 @@
+            "modelAsString": true
+          }
+        },
+        "createdDate": {
```
>createdDate [](http://example.com/codeflow?start=9&length=11)

you can specify the date format

razvanbadea-msft commented on specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2025-02-01/Workspaces.json:
```diff
@@ -0,0 +1,890 @@
+          "description": "The last time when the replication was enabled.",
+          "readOnly": true
+        },
+        "lastModifiedDate": {
```
>lastModifiedDate [](http://example.com/codeflow?start=9&length=16)

you can specify the date format

roizhv22 replied:
addressed

razvanbadea-msft commented on specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2025-02-01/Workspaces.json:
```diff
@@ -0,0 +1,890 @@
+          "readOnly": true
+        }
+      },
+      "required": [
```
>required [](http://example.com/codeflow?start=7&length=8)

this properties are used also for PATCH operations. You can create a new model for patch that contains only patchable operations

razvanbadea-msft commented:
/azp run

roizhv22 committed:
Add format to dateTime fields

Addressing format comments of ARM reviewer

roizhv22 commented on specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2025-02-01/Workspaces.json:
```diff
@@ -0,0 +1,890 @@
+          }
+        },
+        "lastModifiedDate": {
+          "type": "string",
```
addressed

roizhv22 committed:
remove required properties on replication

as asked

roizhv22 committed:
Change to required date-time format

added as required by review

roizhv22 committed:
fix to validate

ms-bogdan-rotaru commented:
The comment on `location` property is not addressed.

roizhv22 commented:
> The comment on `location` property is not addressed.

Hi, I've comment the following, but comment marked as pending
This is the answer
-----

Hi, so this is not a top level property as it's under the properties of the object. It was already agreed in a preview version we've published and customers are already using this.

Here is the example how the contract looks like,

Workspace resource payload
 
"workspace-resource": {
       "location": "westeurope",
 
       "properties": {
           "provisioningState": "succeeded",
 
           "replication": {
 "enabled": true/false, // Enable or disable replication. When disabled, replication to the specified location is stopped and failover is not possible.
 "location": "northeurope",
 
 // Read-only properties
"provisioningState": "Succeeded"/"EnableRequested"/"Enabling"/"DisableRequested"/"Disabling"/"Failed"
               "lastUpdatedTime": "DATETIME" //  Updated after each change in provisioningState
           },
 
           // Read-only property - modified by POST failover/failback
           "failover": {
              "state": "Activating"/"Active"/"Deactivating"/"Inactive"
              "lastUpdateTime": "DATETIME"
           }
       }
   }

ms-bogdan-rotaru commented:
/azp run

ms-bogdan-rotaru commented:
Swagger LintDiff check is failing, can you please check?

roizhv22 commented:
> Swagger LintDiff check is failing, can you please check?

So this API does not returned 201, only 200 and 202. I don't understand why this test is failing if 201 is not present in the contract 
(We've fixed the swagger description to match our service behavior, this why I also added the BreakingChange-Approved-BugFix)

jliusan committed:
add suppression

jliusan committed:
Merge branch 'main' into roizehavi/OiNewStable

kazrael2119 commented:
/azp run

roizhv22 committed:
Merge branch 'main' into roizehavi/OiNewStable

kazrael2119 committed:
Update sdk-suppressions.yaml

roizhv22 committed:
Merge branch 'main' into roizehavi/OiNewStable

roizhv22 committed:
Merge branch 'main' into roizehavi/OiNewStable

roizhv22 commented:
Still in progress

roizhv22 commented:
Comment for activity

roizhv22 committed:
Merge branch 'main' into roizehavi/OiNewStable

roizhv22 commented:
comment

roizhv22 committed:
Merge branch 'main' into roizehavi/OiNewStable

ArcturusZhang commented:
It is fine that we ignore the failure on azure-sdk-for-net pipeline which causes by duplicate schemas.
By checking the log, it comes from the common-types - which is impossible to resolve them here.
We could resolve them and generate proper .net sdk when you open a PR in azure-sdk-for-net via directives.

dulikvor commented:
> It is fine that we ignore the failure on azure-sdk-for-net pipeline which causes by duplicate schemas. By checking the log, it comes from the common-types - which is impossible to resolve them here. We could resolve them and generate proper .net sdk when you open a PR in azure-sdk-for-net via directives.

Could you approve the .net step, otherwise the automation won't continue.
